### ISSUE TITLE
Download all attachments as ZIP is broken

### ISSUE THREAD

**Issue by st3iny** [nextcloud/mail#6755]
### Steps to reproduce

1. Navigate to an envelope with many attachments.
2. Click on "Download Zip" below the attachment list.
3. There won't be a download and you will be redirected to an error page.

![Peek 2022-06-17 12-37](https://user-images.githubusercontent.com/1479486/174283033-f78d53de-4f10-4c40-a6d4-7f0d0ac7dbf8.gif)

### Expected behavior

The browser should start downloading my attachments zip file.

### Actual behavior

There will be now download and I am redirected to an error page.

```json
{
  "status": "error",
  "message": "Class \"Horde_Nls\" not found in file '/srv/http/dev_apps/mail/vendor/bytestream/horde-mime/lib/Horde/Mime/Part.php' line 1317",
  "data": {
    "debug": true,
    "type": "Exception",
    "message": "Class \"Horde_Nls\" not found in file '/srv/http/dev_apps/mail/vendor/bytestream/horde-mime/lib/Horde/Mime/Part.php' line 1317",
    "code": 0,
    "trace": [
      {
        "file": "/srv/http/lib/private/AppFramework/App.php",
        "line": 172,
        "function": "dispatch",
        "class": "OC\\AppFramework\\Http\\Dispatcher"
      },
      {
        "file": "/srv/http/lib/private/Route/Router.php",
        "line": 298,
        "function": "main",
        "class": "OC\\AppFramework\\App"
      },
      {
        "file": "/srv/http/lib/base.php",
        "line": 1023,
        "function": "match",
        "class": "OC\\Route\\Router"
      },
      {
        "file": "/srv/http/index.php",
        "line": 36,
        "function": "handleRequest",
        "class": "OC"
      }
    ],
    "previous": {
      "type": "Error",
      "message": "Class \"Horde_Nls\" not found",
      "code": 0,
      "trace": [
        {
          "file": "/srv/http/dev_apps/mail/lib/IMAP/MessageMapper.php",
          "line": 612,
          "function": "getSize",
          "class": "Horde_Mime_Part"
        },
        {
          "file": "/srv/http/dev_apps/mail/lib/Service/MailManager.php",
          "line": 597,
          "function": "getAttachments",
          "class": "OCA\\Mail\\IMAP\\MessageMapper"
        },
        {
          "file": "/srv/http/dev_apps/mail/lib/Controller/MessagesController.php",
          "line": 575,
          "function": "getMailAttachments",
          "class": "OCA\\Mail\\Service\\MailManager"
        },
        {
          "file": "/srv/http/lib/private/AppFramework/Http/Dispatcher.php",
          "line": 225,
          "function": "downloadAttachments",
          "class": "OCA\\Mail\\Controller\\MessagesController"
        },
        {
          "file": "/srv/http/lib/private/AppFramework/Http/Dispatcher.php",
          "line": 133,
          "function": "executeController",
          "class": "OC\\AppFramework\\Http\\Dispatcher"
        },
        {
          "file": "/srv/http/lib/private/AppFramework/App.php",
          "line": 172,
          "function": "dispatch",
          "class": "OC\\AppFramework\\Http\\Dispatcher"
        },
        {
          "file": "/srv/http/lib/private/Route/Router.php",
          "line": 298,
          "function": "main",
          "class": "OC\\AppFramework\\App"
        },
        {
          "file": "/srv/http/lib/base.php",
          "line": 1023,
          "function": "match",
          "class": "OC\\Route\\Router"
        },
        {
          "file": "/srv/http/index.php",
          "line": 36,
          "function": "handleRequest",
          "class": "OC"
        }
      ],
      "previous": null
    }
  },
  "code": 0
}
```

### Mail app version

main

### PHP engine version

PHP 8.0

### Web server

Apache (supported)

### Database

MariaDB

**Comment by kesselb** [comment]
Horde_Nls is an optional dependency for the Mime package 
https://github.com/horde/Mime/blob/7f6d0aaea1a1f194f46ac4a0428b539df11b46e1/composer.json#L40

but used without any checks in 
https://github.com/horde/Mime/blob/7f6d0aaea1a1f194f46ac4a0428b539df11b46e1/lib/Horde/Mime/Part.php#L1317

**Comment by kesselb** [comment]
> You could open a PR with https://github.com/bytestream/Mime as we're using their fork for PHP8.1 support.

Maybe ;) I'm not sure how to fix it properly. Right now the composer definition is wrong. It's a hard dependency and should be in dependency section. To keep it an optional dependency the code should work without the other package. 

In our case it's the formatting of the bytes that needs the other library. What we could do in mail is to switch from getSize to getBytes and format the value ourself.

**Comment by vipergts450** [comment]
Hello, this fix did not appear to be pushed into the docker container version of nextcloud because my MessageMapper.php at /config/www/nextcloud/apps/mail/lib/IMAP/ shows getSize() on line 609 rather than getBytes().

**Comment by st3iny** [comment]
> Hello, this fix did not appear to be pushed into the docker container version of nextcloud because my MessageMapper.php at /config/www/nextcloud/apps/mail/lib/IMAP/ shows getSize() on line 609 rather than getBytes(). I run into the same error as the original contributor when trying to download all attachments as a ZIP.
> 
> For more info, I am running the latest docker image of nextcloud and the most recent Mail app, so I am not sure where along the way it should have grabbed this change, but it did not, oddly.

The patch was not backported to stable1.13 and thus hasn't been released yet. You can expect it to be included in 1.14.x.

**Comment by vipergts450** [comment]
Got it! Thanks for the explanation. I made the change manually in the
interim.
